### DISCUSSION TITLE
Capabilities and options for categorising and searching the standards?

### DISCUSSION THREAD

**Discussion by willhiorns** [UKHomeOffice-test/docs-as-code-nunjucks#22] [General]
Is there any labelling capability?
What can and can't be surfaced with search?

Creators and consumers of standards probably have some different needs around these questions.

**Comment by edhamiltonHO** [comment]
Labelling is available via tags, haven't looked into the search functionality just yet

**Reply by gstevenson** [reply]
https://github.com/UKHomeOffice-test/docs-as-code-nunjucks/search?q=Capabilities&type=discussions

**Reply by gstevenson** [reply]
You can filter search results by 'type' to narrow down to discussions

**Reply by RobJohnsonHO** [reply]
I've added tags for each of the guilds to mark content/issues/discussions where appropriate.

**Reply by edhamiltonHO** [reply]
Nice @RobJohnson-HO 👍 . We should probably think about the tags we use within nunjucks to manage the content in the site too, at the moment just marking principles as such. One for this thread or should we open a new one?

**Reply by RobJohnsonHO** [reply]
I guess it depends on the OP, did @willhiorns mean searching the static site once generated or GitHub.  The tags I created are in GitHub.  The tags in the site will help searching once its build and a static site.

**Reply by robertdeniszczyc2** [reply]
I recall search was something we initially talked about in the original innovation day, and the counter point was if you need to search for something, is the information laid out in the best way in the first place since we're talking about relatively small amounts of content here.

**Reply by willhiorns** [reply]
> I guess it depends on the OP, did @willhiorns mean searching the static site once generated or GitHub. The tags I created are in GitHub. The tags in the site will help searching once its build and a static site.

I was meaning search of the built product, by end users.

**Reply by willhiorns** [reply]
> I recall search was something we initially talked about in the original innovation day, and the counter point was if you need to search for something, is the information laid out in the best way in the first place since we're talking about relatively small amounts of content here.

My perspective on this is that it's very hard to come up a coherent structure that doesn't yield gaps and duplications.  Moreover, that will tend to be a tree that decomposes according to a single analytical concept.  But the classic problem is that both "vertical" (domain, concrete) vs "horizontal" (technical, abstract) decompositions are absolutely valid and relevant.  So what always happens with any tree of information is that these two get mixed up and the overall categorisation gets corrupted.

Search capabilities are transformative (literally and figuratively).  They turn a static site into something dynamic, akin to MVC separation.  And that puts the user's immediate context into the results, via the search terms.  That abstraction of view from data is **vital** in my opinion.  

Would you be able to use Stackoverflow without a search engine (either external e.g. Google or internal to SO)?

I'm not convinced by the "small amounts of content" argument.  I think that presents a risk of reaching a point where the amount of content accreted makes the site unusable because we lack this key functionality.

So, what's the end user search capability?  And is that augmented with a concept like tagging?

**Comment by willhiorns** [comment]
Note on Discussions.  Linear conversations are a bind once you've used Slack for a bit!  Looks to me like there's no threading, right?

Oh, wait, no don't answer that here!  ;)

**Reply by willhiorns** [reply]
Oh hang on, I've just _accidentally_ started a new comment thread.  How embarrassing!  I guess I could "copy link" rather than "quote reply"?

https://github.com/UKHomeOffice-test/docs-as-code-nunjucks/discussions/22#discussioncomment-2615102
